src/components/PressableWithSecondaryInteraction/index.js

```diff
@@ -1,103 +1,124 @@
-         * Therefore it shifts the element to bring it back to focus.
-         * https://github.com/Expensify/App/issues/14148
+         * @param {contextmenu} e - A right-click MouseEvent.
+         * https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event
          */
-        if (this.props.withoutFocusOnSecondaryInteraction && this.pressableRef) {
-            this.pressableRef.blur();
-        }
-    }
-
-    render() {
-        const defaultPressableProps = _.omit(this.props, ['onSecondaryInteraction', 'children', 'onLongPress']);
-        const inlineStyle = this.props.inline ? styles.dInline : {};
-
-        // On Web, Text does not support LongPress events thus manage inline mode with styling instead of using Text.
-        return (
-            <PressableWithFeedback
-                wrapperStyle={StyleUtils.combineStyles(DeviceCapabilities.canUseTouchScreen() ? [styles.userSelectNone, styles.noSelect] : [], inlineStyle)}
-                onPressIn={this.props.onPressIn}
-                onLongPress={this.props.onSecondaryInteraction ? this.executeSecondaryInteraction : undefined}
-                pressDimmingValue={this.props.activeOpacity}
-                onPressOut={this.props.onPressOut}
-                onPress={this.props.onPress}
-                ref={(el) => (this.pressableRef = el)}
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...defaultPressableProps}
-                style={(state) => [StyleUtils.parseStyleFromFunction(this.props.style, state), inlineStyle]}
-            >
-                {this.props.children}
-            </PressableWithFeedback>
-        );
-    }
+        const executeSecondaryInteractionOnContextMenu = (e) => {
+            if (!onSecondaryInteraction) {
+                return;
+            }
+
+            e.stopPropagation();
+            if (preventDefaultContextMenu) {
+                e.preventDefault();
+            }
+
+            onSecondaryInteraction(e);
+
+            /**
+             * This component prevents the tapped element from capturing focus.
+             * We need to blur this element when clicked as it opens modal that implements focus-trapping.
+             * When the modal is closed it focuses back to the last active element.
+             * Therefore it shifts the element to bring it back to focus.
+             * https://github.com/Expensify/App/issues/14148
+             */
+            if (withoutFocusOnSecondaryInteraction) {
+                element.blur();
+            }
+        };
+
+        element.addEventListener('contextmenu', executeSecondaryInteractionOnContextMenu);
+
+        return () => {
+            element.removeEventListener('contextmenu', executeSecondaryInteractionOnContextMenu);
+        };
+        // We only want this to run on mount
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
Seems like it's safe to remove this and add the necessary dependencies to the effect: `onSecondaryInteraction`, `preventDefaultContextMenu`, `withoutFocusOnSecondaryInteraction`. These props shouldn't change often, right?

The effect might run a couple times more if some of the props aren't stable but seems safe, given that the listener is gonna be cleaned if it re-runs. If possible, I always advocate for leaving the right dependencies otherwise this can cause hard to debug bugs later on when another dev is refactoring this